app/presenters/deploy_group_usage_csv_presenter.rb

```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+require 'csv'
+
+class DeployGroupUsageCsvPresenter
+  def self.to_csv
+    CSV.generate do |csv|
+      csv << csv_header
+      Project.all.each do |project|
+        if project.stages.empty?
+          csv << csv_line(project)
+        else
+          project.stages.each do |stage|
+            stage.deploy_groups.each do |deploy_group|
+              csv << csv_line(project, stage, deploy_group)
+            end
+          end
+        end
+      end
+    end
+  end
+
+  def self.csv_header
+    [
+      "project_name",
+      "stage_name",
+      "deploy_group_name",
+      "deploy_group_environment_id",
+      "deploy_group_environment_name",
+      "deploy_group_env_value",
+      "deploy_group_permalink"
+    ]
+  end
+
+  def self.csv_line(project, stage = nil, deploy_group = nil)
+    result = []
```

Comment:
```Ruby
result = [project.name]
if stage && deploy_group
  result.concat [
    stage.name,
    ...
  ]
end
```

Reply:
I looked at sorta doing it this way - but the eval of the `if statement` blows away the auto return of `result` - so something like this would be necessary

```
def self.csv_line(project, stage = nil, deploy_group = nil)
  result = [project.name]
  if stage && deploy_group
    result.concat [
      stage.name,
      ...
    ]
  end
  result
end
```

It seemed more clever :D to instead flip the logic around and make the last statement (which will always have a good return value) to shove the always present first value into the array. 

But, I guess I am not really saving any lines here, still need to have the init then the return, so I will make the proposed change.

Reply:
done